sycl/source/detail/plugin_printers.hpp

```diff
@@ -19,110 +19,108 @@ namespace sycl {
 namespace detail {
 namespace pi {
 
+
+//Note:Casting val to unsigned long may not work for all function specializations
 template <typename T>
 inline typename std::enable_if<!std::is_pointer<T>::value, void>::type
 print(T val) {
-  std::cout << "<unknown> : " << val << std::endl;
+  printf("<unknown> : ");
+  printf("%lu",(unsigned long)val);
+  printf("\n");
```

Comment:
To my knowledge, this will have an effect on the readability of our traces. @smaslov-intel - Thoughts on how to do this without std::cout?